backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/document/impl/DocTypeBL.java

```diff
@@ -110,4 +86,10 @@ public boolean isPrepay(final I_C_DocType dt)
 		return X_C_DocType.DOCSUBTYPE_PrepayOrder.equals(dt.getDocSubType())
 				&& X_C_DocType.DOCBASETYPE_SalesOrder.equals(dt.getDocBaseType());
 	}
+
+	@Override
+	public boolean hasRequestType(@NonNull final DocTypeId docTypeId)
+	{
+		return Services.get(IDocTypeDAO.class).getById(docTypeId).getR_RequestType_ID() > 0;
```

Comment:
extract the service as a field

Reply:
@teosarca done, also replaced it in other methods from the class with the class field.